### ISSUE TITLE
Can't get toplevel completion to work when completion cache is enabled "Type name _ is redefined from module _"

### ISSUE THREAD

**Issue by jeremyfa** [vshaxe/vshaxe#125]
Not sure if it is a haxe issue or vshaxe issue (and/or if it's a known issue), but as I am experiencing it in vshaxe, here we go:

In my project, toplevel completion fails if completion cache is enabled. This is what vshaxe tells me when failing (example inside a Project.hx file):

```
Haxe language server started
Initializing completion cache...
Done.
[Warn  - 6:22:23 PM] Request textDocument/completion failed.
  Message: src/Project.hx:3: characters 7-21 : Type name Project is redefined from module Project
  Code: -32603 
[Warn  - 6:23:47 PM] Request textDocument/completion failed.
  Message: src/Project.hx:3: characters 7-21 : Type name Project is redefined from module Project
  Code: -32603 
```

Field access completion does work fine, but I need to disable completion cache (and make it very slow) in order to get toplevel completion to work.

Unfortunately I did not manage to reproduce the case on a smaller example, but I believe it happens when macro code gets mixed up with "regular" code in the completion server.

I tried to change a few things in the project, but the same error (for a type, or another) keeps coming back. At an older state of the project, I was getting these errors if AND ONLY if I opened a haxe class containing macro code. But now it happens everywhere, even without opening a class with macro code (but there are still macros involved in the classes I use).

At the moment, I either have the choice between slow completion (without cache) or incomplete completion (without toplevel), so any help is welcome.

**Comment by jeremyfa** [comment]
More info on the subject: in order to understand what triggers the problem, I tested various commits in the history of my project. I found what "triggers" the issue (even though I don't really understand why). I believe there is definitely a bug related to haxe completion server, but I will need help to spot it. The project is private so I can't share it publicly, but I can give access to somebody of the haxe foundation so that he can reproduce the issue.
